pkg/check/permissions.go

```diff
@@ -0,0 +1,326 @@
+		"StorageProfile.View",
+	},
+	permissionCluster: {
+		"Host.Config.Storage",
+		"Resource.AssignVMToPool",
+		"VApp.AssignResourcePool",
+		"VApp.Import",
+		"VirtualMachine.Config.AddNewDisk",
+	},
+	permissionPortgroup: {
+		"Network.Assign",
+	},
+	permissionFolder: {
+		"Resource.AssignVMToPool",
+		"VApp.Import",
+		"VirtualMachine.Config.AddExistingDisk",
+		"VirtualMachine.Config.AddNewDisk",
+		"VirtualMachine.Config.AddRemoveDevice",
+		"VirtualMachine.Config.AdvancedConfig",
+		"VirtualMachine.Config.Annotation",
+		"VirtualMachine.Config.CPUCount",
+		"VirtualMachine.Config.DiskExtend",
+		"VirtualMachine.Config.DiskLease",
+		"VirtualMachine.Config.EditDevice",
+		"VirtualMachine.Config.Memory",
+		"VirtualMachine.Config.RemoveDisk",
+		"VirtualMachine.Config.Rename",
+		"VirtualMachine.Config.ResetGuestInfo",
+		"VirtualMachine.Config.Resource",
+		"VirtualMachine.Config.Settings",
+		"VirtualMachine.Config.UpgradeVirtualHardware",
+		"VirtualMachine.Interact.GuestControl",
+		"VirtualMachine.Interact.PowerOff",
+		"VirtualMachine.Interact.PowerOn",
+		"VirtualMachine.Interact.Reset",
+		"VirtualMachine.Inventory.Create",
+		"VirtualMachine.Inventory.CreateFromExisting",
+		"VirtualMachine.Inventory.Delete",
+		"VirtualMachine.Provisioning.Clone",
+		"VirtualMachine.Provisioning.MarkAsTemplate",
+		"VirtualMachine.Provisioning.DeployTemplate",
+	},
+	permissionDatacenter: {
+		"Resource.AssignVMToPool",
+		"VApp.Import",
+		"VirtualMachine.Config.AddExistingDisk",
+		"VirtualMachine.Config.AddNewDisk",
+		"VirtualMachine.Config.AddRemoveDevice",
+		"VirtualMachine.Config.AdvancedConfig",
+		"VirtualMachine.Config.Annotation",
+		"VirtualMachine.Config.CPUCount",
+		"VirtualMachine.Config.DiskExtend",
+		"VirtualMachine.Config.DiskLease",
+		"VirtualMachine.Config.EditDevice",
+		"VirtualMachine.Config.Memory",
+		"VirtualMachine.Config.RemoveDisk",
+		"VirtualMachine.Config.Rename",
+		"VirtualMachine.Config.ResetGuestInfo",
+		"VirtualMachine.Config.Resource",
+		"VirtualMachine.Config.Settings",
+		"VirtualMachine.Config.UpgradeVirtualHardware",
+		"VirtualMachine.Interact.GuestControl",
+		"VirtualMachine.Interact.PowerOff",
+		"VirtualMachine.Interact.PowerOn",
+		"VirtualMachine.Interact.Reset",
+		"VirtualMachine.Inventory.Create",
+		"VirtualMachine.Inventory.CreateFromExisting",
+		"VirtualMachine.Inventory.Delete",
+		"VirtualMachine.Provisioning.Clone",
+		"VirtualMachine.Provisioning.DeployTemplate",
+		"VirtualMachine.Provisioning.MarkAsTemplate",
+		"Folder.Create",
+		"Folder.Delete",
+	},
+	permissionDatastore: {
+		"Datastore.AllocateSpace",
+		"Datastore.Browse",
+		"Datastore.FileManagement",
+		"InventoryService.Tagging.ObjectAttachable",
+	},
+}
+
+func comparePrivileges(ctx context.Context, username string, mo types.ManagedObjectReference, authManager AuthManager, required []string) error {
+	derived, err := authManager.FetchUserPrivilegeOnEntities(ctx, []types.ManagedObjectReference{mo}, username)
```

Comment:
Is there any special privilege required for fetching user privileges on entities? Or all users are entitled for this permission?

Reply:
This call works for an authenticated user with no stated privileges.   I tested this with a user which held only read-only privileges and the call succeeded.  Additionally, the [vCenter SOAP API doc](https://developer.vmware.com/apis/358/vsphere/doc/vim.AuthorizationManager.html) doesn't state a requirement for a specific privilege to be held.